teflo/core.py

```
@@ -397,13 +398,22 @@ def __init__(self, config=None, name=None, **kwargs):
 
         # every resource can have optional labels
         self._labels = list()
+        self._resource_id = gen_random_str(10)
 
     def __eq__(self, o) -> bool:
-        return self.name == o.name
+        return self.resource_id == o.resource_id
 
     def __hash__(self) -> int:
         return super().__hash__()
 
+    @property
+    def resource_id(self):
+        return self._resource_id
+
+    @resource_id.setter
```

i realized this when i saw your unit test, we should not be setting resource_id. It should be auto generated. So the setter should throw an error stating resource_id cannot be set.
Something like this 
`    @groups.setter
    def groups(self, value):
        """Set groups property."""
        raise AttributeError('You cannot set the groups after asset class is '
                             'instantiated.')`

Also this will force you to change the unit test where you are setting the id.

---

Why should we forbid this?

---

resource id should not be changed in between, it is a unique id and the system should generate this. it is not something the user should be able to provide or the program to alter in between